spec/rspec/mocks/matchers/receive_spec.rb

```diff
@@ -438,23 +444,23 @@ def receiver.method_missing(*a); end # a poor man's stub...
         end
 
         it 'supports `expect(...).not_to receive`' do
-          dbl = double
+          with_unfulfilled_double do |dbl|
+            framework.new.instance_exec do
+              expect(dbl).not_to receive(:foo)
+            end
 
-          framework.new.instance_exec do
-            expect(dbl).not_to receive(:foo)
+            expect { dbl.foo }.to raise_error(RSpec::Mocks::MockExpectationError)
           end
-
-          expect { dbl.foo }.to raise_error(RSpec::Mocks::MockExpectationError)
         end
 
         it 'supports `expect(...).to_not receive`' do
-          dbl = double
+          with_unfulfilled_double do |dbl|
```

Comment:
I think `expect_fast_failure_from` could take the place of `with_unfulfilled_double` in these situations as well -- you could have it yield the double to the block.